app/scripts/lib/rpc-method-middleware/handlers/ethereum-chain-utils.js

```diff
@@ -165,20 +165,20 @@ export function validateAddEthereumChainParams(params) {
  * @param {string} chainId - The chainId being switched to.
  * @param {string} networkClientId - The network client being switched to.
  * @param {object} hooks - The hooks object.
- * @param {boolean} hooks.isAddFlow - The boolean determining if this call originates from wallet_addEthereumChain.
  * @param {Function} hooks.setActiveNetwork - The callback to change the current network for the origin.
  * @param {Function} hooks.getCaveat - The callback to get the CAIP-25 caveat for the origin.
  * @param {Function} hooks.requestPermittedChainsPermissionForOrigin - The callback to request a new permittedChains-equivalent CAIP-25 permission.
  * @param {Function} hooks.requestPermittedChainsPermissionIncrementalForOrigin - The callback to add a new chain to the permittedChains-equivalent CAIP-25 permission.
- * @returns a null response on success or an error if user rejects an approval when isAddFlow is false or on unexpected errors.
+ * @param hooks.autoApprove
```

Comment:
[nit] not your doing (probably my doing). this should probably not be called the hooks object....

Reply:
```suggestion
 * @param hooks.autoApprove - a boolean indicating whether the request should prompt the user or be automatically approved.
```

Reply:
mmm, i think while it isn't quite a hook, it should exist in that param object

Reply:
https://github.com/MetaMask/metamask-extension/pull/29837/commits/e5f13fba1834b88020b25277d9f3000339f5da64